Pre-release r3.1 for Fall25 m3 milestone

jpengar opened the pull request:
#### What type of PR is this?

* repository management

#### What this PR does / why we need it:

- creates the API readiness checklist
- creates and adds section for r3.1 to the changelog
- update README.md
- `PENDING` updates the versions and server URLs in API and test definition files
- `PENDING` updates the API spec according to CAMARA ICM r3.2 and CAMARA Commonalities r3.2.

#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Fixes #24

#### Special notes for reviewers:

The PR is set to 'DRAFT' for the time being, until PRs #20 and #22 have been resolved.

#### Changelog input

```
 Pre-release r3.1 for Fall25 m3 milestone
```

#### Additional documentation 

N/A

hdamker commented:
@jpengar Just seen now: why r3.1? It's the very first release in this repository, so you should start with r1.1

jpengar commented:
> > * `PENDING` adds basic test definitions (**NOTE: This may need to wait until public release**)
> 
> Please try to add them still as soon as possible ... if possible before the creation of the release candidate. They are mandatory to give implementations of the rc some test cases at hand.

I would love to, but first we need to finalise the API design, particularly the content of the API response, in order to properly define the test scenarios 😕

jpengar commented:
> @jpengar Just seen now: why r3.1? It's the very first release in this repository, so you should start with r1.1

My bad. I'll fix it.

diegogonmar reviewed (approved):
LGTM

jpengar commented:
> @jpengar Great that this PR is ready for review!
>
> A few minor and easy to fix issues found in the API and test definition files:

@hdamker "Haste makes waste." 😓. Thank you for the detailed review. Hopefully, I have addressed all your comments and suggestions in https://github.com/camaraproject/ConsentInfo/pull/25/commits/1987f9cf58b5d6f4597b0a48c1c874fa2d9c5230

hdamker commented:
> Hopefully, I have addressed all your comments and suggestions in [1987f9c](https://github.com/camaraproject/ConsentInfo/commit/1987f9cf58b5d6f4597b0a48c1c874fa2d9c5230)

@jpengar looks good on first glance, I will come back later for a final review. At least our [robot](https://github.com/hdamker/ReleaseManagement/issues/29#issuecomment-3051784193) is now happy (but has only partial coverage): 

🔴 Critical: 0
🟡 Medium: 0
🔵 Low: 0

jpengar commented:
@hdamker I'm afraid I'll be out of office next week. I apologise for bothering you, as I know you're very busy at the moment. Do you think we could finalise the review of this PR to generate the release before I leave? I just want to avoid any problems as I am the PR owner.

hdamker commented:
> @hdamker I'm afraid I'll be out of office next week. I apologise for bothering you, as I know you're very busy at the moment. Do you think we could finalise the review of this PR to generate the release before I leave? I just want to avoid any problems as I am the PR owner.

I don't think that there will be further comments from me and ReleaseManagement side. So if there are the approvals from the other codeowners they can also merge it.

sebdewet reviewed (approved):
LGTM
